### PR TITLE
tree: restore the isHeld task status attribute

### PR DESCRIPTION
**Pull request by oliver-sanders**
**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [ ] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

**Comment by hjoliver** [comment]
Presumably this is a temporary change.  Can we have an explanation @oliver-sanders ?

**Comment by oliver-sanders** [comment]
> Presumably this is a temporary change. Can we have an explanation @oliver-sanders ?

Explanation? This restores the pause symbol which appears next to tasks when they are held. At some point this was lost from the tree view, this PR puts it back.

**Comment by oliver-sanders** [comment]
> Three e2e/functional tests failing for the tree component

I'm guessing the snapshot used for testing the tree view doesn't have the `isHeld` field which is causing traceback resulting in missing nodes. I'm guessing this as cypress is being pretty useless for me ATM.

@kinow I've lost track of where we are at with data snapshots. How do we update these, are we still using `./generate`?

**Comment by kinow** [comment]
>@kinow I've lost track of where we are at with data snapshots. How do we update these, are we still using ./generate?

Yup. The only difference is that there is no more query in `get_checkpoint.py`. Instead it now [parses the JS variable](https://github.com/cylc/cylc-ui/blob/d26d0e9abcea69ebb29637fd672359609801f3a1/src/services/mock/checkpoint/get_checkpoint.py#L8) from the tree component query in JS.

So in theory all you'd need to do is run the `get_checkpoint.py` and update the branch/PR (:crossed_fingers:)

**Comment by hjoliver** [comment]
> Explanation? ...

**Comment by oliver-sanders** [comment]
> Ah, OK, I kind of figured we were partially transitioned to the new task states. But no biggie

We are using the `is_held` state attribute (the old held state went away long ago) misleading title, apologies.

**Comment by oliver-sanders** [comment]
Done,

I've also streamlined the checkpointing process a little and aliased it in the `package.json` (so that I don't forget again).

```
npm run checkpoint
```

File: package.json [hunk 1]
```diff
@@ -4,13 +4,14 @@
   "private": true,
   "license": "GPL-3.0-only",
   "scripts": {
-    "serve": "vue-cli-service serve",
     "build": "vue-cli-service build",
     "build:report": "vue-cli-service build --report",
     "build:watch": "vue-cli-service build --watch --mode development",
-    "lint": "vue-cli-service lint",
+    "checkpoint" : "./src/services/mock/generate",
```
Comment (kinow) [inline package.json:10]: Brilliant idea!

**Comment by oliver-sanders** [comment]
fixing...

File: src/services/mock/generate [hunk 2]
```diff
@@ -1,26 +1,29 @@
 #!/bin/bash
 
+if ! sed --version 2>/dev/null | grep -q 'GNU' 2>/dev/null; then
+    echo -e '** GNU sed required **\n' >&2
```
Comment (oliver-sanders) [inline src/services/mock/generate:4]: POSIX `sed` sucks!

File: src/services/mock/generate [hunk 3]
```diff
@@ -1,26 +1,29 @@
...
-# Finally we export it so we can use it from JS modules
-echo -e '\nexport { checkpoint }' >> "${CHECK_FILE}"
+sed -i \
+    -e "1s|^|const checkpoint = |" \
+    -e "1s|^|/* eslint-disable */\n|" \
+    -e "s|$USER|user|g" \
+    -e "s|$(hostname)|localhost|g" \
+    -e "s|$(hostname | tr '[:upper:]' '[:lower:]')|localhost|g" \
+    -e "\$s|$|\nexport { checkpoint }|" \
+    "${CHECK_FILE}"
```
Comment (kinow) [inline src/services/mock/generate:29]: :rocket: nice! Didn't think in combining everything. Hopefully will remember it for next time, but easier to read.

**Comment by kinow** [comment]
I thought the Yarn one would be simple, but it looks like it doesn't work on macos. Good idea to create a separate one, doing that right now :+1:

**Pull request merged**